src/components/renderers/renderers.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * This module manages registration of renderers for a given sensor type.
+ *
+ * some concepts:
+ * renderers are optional - the system will try to do the best it can. simple numeric values and strings are
+ * rendered. Complex types are rendered as a composite that renders name/value pairs 
+ * todo - extend `spec` to describe sensorType for each element in a complex type?
+ * probably easier to define a group that the readings are in, and these are handled together as a composite.
```

Comment:
Consider it an optimization - the system will attempt to display the data but it may not be optimal. Developers can choose to add their own renderer if they are not happy with the default output.